homeassistant/components/envisalink/services.yaml

```diff
@@ -1,25 +1,47 @@
 # Describes the format for available Envisalink services.
 
 alarm_keypress:
+  name: Alarm keypress
   description: Send custom keypresses to the alarm.
   fields:
     entity_id:
+      name: Entity
       description: Name of the alarm control panel to trigger.
+      required: true
       example: "alarm_control_panel.downstairs"
+      selector:
+        entity:
+          integration: envisalink
+          domain: alarm_control_panel
     keypress:
+      name: Keypress
       description: "String to send to the alarm panel (1-6 characters)."
+      required: true
       example: "*71"
+      selector:
+        text:
 
 invoke_custom_function:
+  name: Invoke custom function
   description: >
     Allows users with DSC panels to trigger a PGM output (1-4).
     Note that you need to specify the alarm panel's "code" parameter for this to work.
   fields:
     partition:
+      name: Partition
       description: >
         The alarm panel partition to trigger the PGM output on.
         Typically this is just "1".
+      required: true
       example: "1"
+      selector:
+        text:
     pgm:
```

Comment:
`pgm` doesn't seem to be part of this service scheam?

Reply:
````ATTR_CUSTOM_FUNCTION = "pgm"
ATTR_PARTITION = "partition"

SERVICE_SCHEMA = vol.Schema(
    {
        vol.Required(ATTR_CUSTOM_FUNCTION): cv.string,
        vol.Required(ATTR_PARTITION): cv.string,
    }
)